src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -710,6 +741,47 @@ private IncrementalHash PrepareDigest()
             }
         }
 
+        private bool VerifySignature(
+            CmsSignature signatureProcessor,
+            X509Certificate2 certificate,
+            bool compatMode)
+        {
+            IncrementalHash hasher = PrepareDigest(compatMode);
+
+            if (hasher == null)
+            {
+                Debug.Assert(compatMode, $"{nameof(PrepareDigest)} returned null for the primary check");
+                return false;
+            }
+
+#if netcoreapp
+            // SHA-2-512 is the biggest digest type we know about.
```

Comment:
Is there anything to assert here?  I'm wondering what would break if in the future a bigger digest type came along.